msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/MzMLSpectrum.java

```diff
@@ -181,7 +208,12 @@ public Integer getNumberOfDataPoints() {
   @Override
   public double[] getMzValues() {
     double[] mzValues = null;
-    this.mzArrayLength = getMzBinaryDataInfo().getArrayLength();
+    Integer numOfDataPoints = this.numOfDataPoints;
```

Comment:
I think there's no need to create Integer numOfDataPoints, since you can just use this.numOfDataPoints